### PR TITLE
PETSc corrections

### PR DESCRIPTION
**Pull request by endJunction**
Collection of unrelated changes for PETSc code.

The MatSetType and Finalize do correct some bugs when using the matrix/vector.
The PETSc includes are marked as system includes and do not produce warnings while compilation.

### CODE REVIEW

**Comment by bilke** [comment]
:+1:

**Comment by norihiro-w** [comment]
:+1:

File: MathLib/LinAlg/PETSc/PETScLinearSolver.h [hunk 1]
```diff
@@ -47,6 +49,9 @@ class PETScLinearSolver
 
         ~PETScLinearSolver()
         {
+            INFO("info: Time elapsed in PETSc ksp solver for equation: %g s.\n",
```
Comment (TomFischer) [inline MathLib/LinAlg/PETSc/PETScLinearSolver.h]: The macro INFO already outputs the string "info: " as default behaviour.

**Comment by TomFischer** [comment]
Apart from tiny remark; :+1:

File: MathLib/LinAlg/PETSc/PETScLinearSolver.h (hunk 1 above)
Comment (wenqing) [inline MathLib/LinAlg/PETSc/PETScLinearSolver.h]: @TomFischer Thanks.

**Pull request merged**